main.go

```diff
@@ -172,6 +176,24 @@ func main() {
 			"reconciliation for InferenceService, please provide --model-registry-inference-reconcile flag.")
 	}
 
+	kserveWithMeshEnabled, kserveWithMeshEnabledErr := utils.VerifyIfComponentIsEnabled(context.TODO(), mgr.GetClient(), utils.KserveAuthorinoComponent)
+	if kserveWithMeshEnabledErr != nil {
+		setupLog.Error(kserveWithMeshEnabledErr, "could not determine if kserve have service mesh enabled")
+	}
+
+	if kserveWithMeshEnabled {
+		ksvcValidatorWebhookSetupErr := builder.WebhookManagedBy(mgr).
+			For(&knservingv1.Service{}).
+			WithValidator(webhook.NewKsvcValidator(mgr.GetClient())).
+			Complete()
+		if ksvcValidatorWebhookSetupErr != nil {
+			setupLog.Error(err, "unable to setup Knative Service validating Webhook")
+			os.Exit(1)
+		}
+	} else {
+		setupLog.Info("Skipping setup of Knative Service validating Webhook.")
```

Comment:
If you really have to log such info perhaps it's good to add "why" it is skipped.

Reply:
I agree. I'll add a reason.

Reply:
Fixed... That said, I'm no longer sure how much we can skip this. 
The conditional is in prevention of RawDeployment setup, where Knative is missing. However, AFAIK, there is no way to introduce equivalent conditionals around the `ValidatingWebhookConfiguration` resource.

So, if the user has Knative for other uses, the `ValidatingWebhookConfiguration` would take effect and would break the cluster, given the controller would have the logic turned off.

I'll tag @VaishnaviHire and @zdtsw for their advice.